lib/applications/routes.js

```diff
@@ -4,53 +4,322 @@
+const uuid = require('uuid');
+
+const database = require('../database');
+const model = require('./model');
+const utils = require('../utils');
 
 /**
  * All Webhooks Landing page
  * @param {object} req - HTTP Request
  * @param {object} res - HTTP Response
  * @param {object} next - Express callback
  *
+
+ * @returns {promise} promise wrapping database promise
  */
-const all = () => {};
+const all = (req, res, next) => {
+  return new Promise((resolve, reject) => {
+    return database
+      .select('*')
+      .from(`${config.applications.base}`)
+      .orderBy('name', 'DESC').then(rws => {
+        // add itentifier to each row
+        const rows = utils.routes.identifier(rws, req.applications.merged);
+        const applications = rows.map(rw => {
+          const row = rw;
+          row.client = {
+            id: row['client-id'],
+            secret: row['client-secret'],
+          };
+
+          return row;
+        });
+
+        res.render('applications/all', {
+          applications,
+          config: config.applications,
+          structure: model.structure,
+        });
+
+        resolve(true);
+      }).catch(e => {
+        const err = {
+          message: e.message,
+          safe: '/',
+          status: 500,
+        };
+
+        reject(err);
+      });
+  }).catch(err => {
+    return next(err);
+  });
+};
 
 /**
  * Add a new Webhook
  * @param {object} req - HTTP Request
  * @param {object} res - HTTP Response
  * @param {object} next - Express callback
  *
+ * @returns {promise} content-types promise
  *
  * TODO: Content-Types errors need to be vastly expanded to properly use promise catch for error
  * https://github.com/punchcard-cms/content-types/issues/112
  */
-const add = () => {};
+const add = (req, res, next) => {
+  const errors = _.get(req.session, 'form.applications.save.errors', {});
+  const values = _.get(req.session, 'form.applications.save.content', {});
+
+  _.unset(req.session, 'form.applications.save');
+
+  return types.only(config.applications.base, values, [req.applications.merged], config).then(merged => {
+    return types.form(merged, errors, config).then(form => {
+      res.render('applications/add', {
+        form,
+        action: req.url.replace(config.applications.actions.add, config.applications.actions.save),
+        config: config.applications,
+        structure: model.structure,
+        button: config.applications.actions.save,
+      });
+    });
+  }).catch(e => {
+    const err = {
+      message: e.message,
+      safe: `/${config.applications.base}`,
+      status: 500,
+    };
+
+    return next(err);
+  });
+};
 
 /**
  * View/edit an individual Webhook
  * @param {object} req - HTTP Request
  * @param {object} res - HTTP Response
  * @param {object} next - Express callback
  *
+ * @returns {promise} resolves/rejects promise
  */
-const one = () => {};
+const one = (req, res, next) => {
+  const id = _.get(req.session, 'form.applications.edit.id', null);
```

Comment:
no, this id is specifically coming from session (or null). It is set in `save` and used here if save rejects back to `one`.